### PR TITLE
Dse/abba prod 13112

### PR DESCRIPTION
**Pull request by sellisd**
Basic abba functionality added to babylon. Input is one csv file with a table of the different simulation runs and their parameters and a corresponding file with the parameter data types. Output is a simple html document with the performance statistics. The command is run in two parts first `babylon abba run` and then `babylon abba check`, the first one launches the runs and the second checks if they are ready and generates the summary

### CODE REVIEW

File: Babylon/commands/abba/run.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    return CommandResponse.success({'rows': rows})
+
+
+@command()
+@injectcontext()
+@pass_azure_token("csm_api")
+@retrieve_state
+@argument("input", type=File('r'))
+@argument("var_types", type=File('r'))
+def check(state: Any, azure_token: str, input: str, var_types: str) -> CommandResponse:
```
Comment (esasova) [inline Babylon/commands/abba/run.py:112]: Could you please put this command in a separate file ? :slightly_smiling_face: all other commands in babylon are separated in different files, we should stick to this convention :slightly_smiling_face:

File: Babylon/commands/abba/run.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        service_state["api"]["workspace_id"] = entry.get("workspaceId")
+        spec = dict()
+        spec["payload"] = json.dumps(entry)
+        scenario_service = ScenarioService(state=service_state, azure_token=azure_token, spec=spec)
+        response = scenario_service.create()
+        if response is None:
+            logger.error(f"Creating {entry['name']} failed")
+            continue
+        scenario = response.json()
+        logger.info(f"Scenario {scenario['id']} creation was posted")
```
Comment (esasova) [inline Babylon/commands/abba/run.py:91]: Can we rephrase it to something like _Scenario has been created_ ? :slightly_smiling_face:

File: Babylon/commands/abba/run.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        df.loc[i, 'scenarioId'] = scenario['id']
+        # run scenario
+        scenario_service.state["api"]["scenario_id"] = scenario["id"]
+        response = scenario_service.run()
+        if response is None:
+            logger.error(f"Failed run {scenario['id']}")
+            continue
+        scenario_run = response.json()
+        df.loc[i, 'scenariorunId'] = scenario_run['id']
+        logger.info(f"Scenario {scenario_run['id']} run posted")
```
Comment (esasova) [inline Babylon/commands/abba/run.py:101]: The same thing, can we rephrase it ? :slightly_smiling_face:

File: Babylon/commands/abba/run.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    return response.json()
+
+
+def summarize(data: dict, i: int):
+    df = pd.DataFrame.from_records(data['nodes'])[['containerName', 'startTime', 'endTime']]
+    df['startTime'] = pd.to_datetime(df['startTime'])
+    df['endTime'] = pd.to_datetime(df['endTime'])
+    df['duration'] = (df['endTime'] - df['startTime']).dt.total_seconds()
+    logger.info(f"Report {i} generated")
+    return (df)
```
Comment (esasova) [inline Babylon/commands/abba/run.py:158]: unnecessary parenthesis :slightly_smiling_face:

File: Babylon/commands/abba/run.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+  <!DOCTYPE html>
+  <html>
+  <head>
+    <title>Simulation Report</title>
+  </head>
+  <body>
+    <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
+    <h1>Total execution time</h1>
+    {top}
+    <h1>Execution time by step</h1>
```
Comment (esasova) [inline Babylon/commands/abba/run.py:187]: This is just front-end dev speaking :rofl: we cannot have two h1 tags in one html tree :slightly_smiling_face:

File: Babylon/commands/abba/run.py [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    """Run a series of simulations
+
+    Args:
+      input (str): Table with details of the simulations
+      var_types (str): A table declaring the variable types of columns in input
+    """
+    rows = []  # an array of dicts one for each input row
+    df = pd.read_csv(input, sep='\t')
+    input_types = dict()
+    input_types = {k: v for k, v in (line.rstrip().split("\t") for line in var_types)}
```
Comment (esasova) [inline Babylon/commands/abba/run.py:74]: The same, the declaration on line 73 seems redundant (the same on lines 119-120)

File: Babylon/commands/abba/check.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+def get_scenariorun_status(service_state, azure_token):
+    # get scenariorun status
+    service = ScenarioRunService(state=service_state, azure_token=azure_token)
+    response = service.status()
+    if response is None:
+        logger.error(f"Failed to get status of {service_state['api']['scenariorun_id']}")
+    return response.json()
+
+
+def summarize(data: dict, i: int):
```
Comment (mahkoCosmo) [inline Babylon/commands/abba/check.py:26]: please use more explicit parameter name

File: Babylon/commands/abba/check.py [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        file.write(html_content)
+
+
+@command()
+@injectcontext()
+@pass_azure_token("csm_api")
+@retrieve_state
+@argument("input", type=File('r'))
+@argument("var_types", type=File('r'))
+def check(state: Any, azure_token: str, input: str, var_types: str) -> CommandResponse:
```
Comment (mahkoCosmo) [inline Babylon/commands/abba/check.py:76]: Maybe a more explicit command's name, 'check_run_status'?

Comment (sellisd) [reply Babylon/commands/abba/check.py:76]: since this is a command in the group abba I thought it might be better to keep it short. The user will be typing `babylon abbba run check` to check the run of abba.

**Review by esasova — APPROVED** [review]
LGTM :+1:
